request-deadline-decorator.js

```diff
@@ -0,0 +1,18 @@
+exports.version = 1;
+
+exports.druidRequestDecoratorFactory = function (logger, params) {
+  return function (decoratorRequest, context) {
+    var query = Object.assign({}, decoratorRequest.query)
+
+    if (context && context.requestDeadline) {
+      var now = new Date().valueOf()
+      if (now > context.requestDeadline) {
+        throw new Error('request deadline overstepped')
+      }
+    }
+
+    return {
+      query: query,
```

Comment:
There shouldn't be any need to return the query here. If you don't return anything it will just take the default options